File: vortexdbg-api/src/main/java/com/vortexdbg/hook/InlineHook.java
package com.vortexdbg.hook;

import com.vortexdbg.Svc;
import com.vortexdbg.Symbol;

public interface InlineHook {

    void replace(long functionAddress, ReplaceCallback callback);
    void replace(Symbol symbol, ReplaceCallback callback);

    void replace(long functionAddress, Svc replace);
    void replace(Symbol symbol, Svc replace);

    void replace(long functionAddress, ReplaceCallback callback, boolean enablePostCall);
    void replace(Symbol symbol, ReplaceCallback callback, boolean enablePostCall);

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/hook/InlineHook.kt
package com.vortexdbg.hook

import com.vortexdbg.Svc
import com.vortexdbg.Symbol

interface InlineHook {

    fun replace(functionAddress: Long, callback: ReplaceCallback)
    fun replace(symbol: Symbol, callback: ReplaceCallback)

    fun replace(functionAddress: Long, replace: Svc)
    fun replace(symbol: Symbol, replace: Svc)

    fun replace(functionAddress: Long, callback: ReplaceCallback, enablePostCall: Boolean)
    fun replace(symbol: Symbol, callback: ReplaceCallback, enablePostCall: Boolean)

}
